File: go/internal/ui/empty_state.go
package ui

import (
	"fmt"
	"strings"
)

// EmptyState represents an empty state component
type EmptyState struct {
	Compact              bool
	Title                string
	SVGPath              string
	PrimaryButtonText    string
	PrimaryButtonLink    string
	PrimaryButtonOptions map[string]interface{}
	SecondaryButtonText  string
	SecondaryButtonLink  string
	SecondaryButtonOptions map[string]interface{}
	EmptyStateOptions    map[string]interface{}
	Description          string
}

// NewEmptyState creates a new empty state with the given parameters
func NewEmptyState(compact bool, title string, svgPath string, primaryButtonText string, primaryButtonLink string, primaryButtonOptions map[string]interface{}, secondaryButtonText string, secondaryButtonLink string, secondaryButtonOptions map[string]interface{}, emptyStateOptions map[string]interface{}, description string) *EmptyState {
	if primaryButtonOptions == nil {
		primaryButtonOptions = make(map[string]interface{})
	}
	if secondaryButtonOptions == nil {
		secondaryButtonOptions = make(map[string]interface{})
	}
	if emptyStateOptions == nil {
		emptyStateOptions = make(map[string]interface{})
	}

	return &EmptyState{
		Compact:              compact,
		Title:                title,
		SVGPath:              svgPath,
		PrimaryButtonText:    primaryButtonText,
		PrimaryButtonLink:    primaryButtonLink,
		PrimaryButtonOptions: primaryButtonOptions,
		SecondaryButtonText:  secondaryButtonText,
		SecondaryButtonLink:  secondaryButtonLink,
		SecondaryButtonOptions: secondaryButtonOptions,
		EmptyStateOptions:    emptyStateOptions,
		Description:          description,
	}
}

// GetEmptyStateClass returns the CSS class for the empty state
func (e *EmptyState) GetEmptyStateClass() string {
	if e.Compact {
		return "gl-flex-row"
	}
	return "gl-text-center gl-flex-col"
}

// GetImageClass returns the CSS class for the image
func (e *EmptyState) GetImageClass() string {
	if e.Compact {
		return "gl-hidden sm:gl-block gl-px-4"
	}
	return "gl-max-w-full"
}

// GetContentWrapperClass returns the CSS class for the content wrapper
func (e *EmptyState) GetContentWrapperClass() string {
	if e.Compact {
		return "gl-grow gl-basis-0 gl-px-4"
	}
	return "gl-m-auto gl-p-5"
}

// GetTitleClass returns the CSS class for the title
func (e *EmptyState) GetTitleClass() string {
	if e.Compact {
		return "h5"
	}
	return "h4"
}

// GetButtonWrapperClass returns the CSS class for the button wrapper
func (e *EmptyState) GetButtonWrapperClass() string {
	if e.Compact {
		return ""
	}
	return "gl-justify-center"
}

// Render generates the HTML for the empty state
func (e *EmptyState) Render() string {
	var parts []string

	// Add empty state class to empty state options
	emptyStateClass := e.GetEmptyStateClass()
	if _, ok := e.EmptyStateOptions["class"]; !ok {
		e.EmptyStateOptions["class"] = fmt.Sprintf("gl-flex gl-empty-state %s", emptyStateClass)
	} else {
		class := e.EmptyStateOptions["class"].(string)
		if !strings.Contains(class, "gl-flex") {
			e.EmptyStateOptions["class"] = fmt.Sprintf("gl-flex gl-empty-state %s %s", emptyStateClass, class)
		}
	}

	// Start empty state section
	parts = append(parts, fmt.Sprintf(`<section %s>`, formatAttributes(e.EmptyStateOptions)))

	// Add SVG image if present
	if e.SVGPath != "" {
		imageClass := e.GetImageClass()
		parts = append(parts, fmt.Sprintf(`<div class="%s"><img src="%s" alt="" class="gl-dark-invert-keep-hue"></div>`, imageClass, e.SVGPath))
	}

	// Add content wrapper
	contentWrapperClass := e.GetContentWrapperClass()
	parts = append(parts, fmt.Sprintf(`<div class="gl-empty-state-content gl-mx-auto gl-my-0 %s">`, contentWrapperClass))

	// Add title
	titleClass := e.GetTitleClass()
	parts = append(parts, fmt.Sprintf(`<h1 class="gl-text-size-h-display gl-leading-36 gl-mt-0 gl-mb-0 %s">%s</h1>`, titleClass, e.Title))

	// Add description if present
	if e.Description != "" {
		parts = append(parts, fmt.Sprintf(`<p class="gl-mt-4 gl-mb-0" data-testid="empty-state-description">%s</p>`, e.Description))
	}

	// Add buttons if present
	if e.PrimaryButtonText != "" || e.SecondaryButtonText != "" {
		buttonWrapperClass := e.GetButtonWrapperClass()
		parts = append(parts, fmt.Sprintf(`<div class="gl-flex gl-flex-wrap gl-mt-5 gl-gap-3 %s">`, buttonWrapperClass))

		// Add primary button if present
		if e.PrimaryButtonText != "" {
			primaryButtonOptions := make(map[string]interface{})
			for k, v := range e.PrimaryButtonOptions {
				primaryButtonOptions[k] = v
			}
			primaryButtonOptions["class"] = "!gl-ml-0"

			primaryButton := NewButton(
				ButtonCategoryPrimary,
				ButtonVariantConfirm,
				ButtonSizeMedium,
				ButtonTypeButton,
				false, // disabled
				false, // loading
				false, // block
				false, // label
				false, // selected
				"", // icon
				e.PrimaryButtonLink, // href
				false, // form
				"", // target
				"", // method
				primaryButtonOptions, // button options
				"", // button text classes
				"", // icon classes
				"", // icon content
				e.PrimaryButtonText, // content
			)
			parts = append(parts, primaryButton.Render())
		}

		// Add secondary button if present
		if e.SecondaryButtonText != "" {
			secondaryButtonOptions := make(map[string]interface{})
			for k, v := range e.SecondaryButtonOptions {
				secondaryButtonOptions[k] = v
			}
			if e.PrimaryButtonText == "" {
				secondaryButtonOptions["class"] = "!gl-ml-0"
			}

			secondaryButton := NewButton(
				ButtonCategoryPrimary,
				ButtonVariantDefault,
				ButtonSizeMedium,
				ButtonTypeButton,
				false, // disabled
				false, // loading
				false, // block
				false, // label
				false, // selected
				"", // icon
				e.SecondaryButtonLink, // href
				false, // form
				"", // target
				"", // method
				secondaryButtonOptions, // button options
				"", // button text classes
				"", // icon classes
				"", // icon content
				e.SecondaryButtonText, // content
			)
			parts = append(parts, secondaryButton.Render())
		}

		// Close button wrapper
		parts = append(parts, "</div>")
	}

	// Close content wrapper
	parts = append(parts, "</div>")

	// Close empty state section
	parts = append(parts, "</section>")

	return strings.Join(parts, "\n")
}

File: rust_app/src/components/pajamas/empty_state.rs
use super::button::ButtonComponent;
use std::collections::HashMap;

/// Renders a Pajamas compliant empty state
#[derive(Debug, Clone)]
pub struct EmptyStateComponent {
    /// Whether the empty state is compact
    compact: bool,
    /// The title of the empty state
    title: Option<String>,
    /// The SVG path for the illustration
    svg_path: Option<String>,
    /// The primary button text
    primary_button_text: Option<String>,
    /// The primary button link
    primary_button_link: Option<String>,
    /// The primary button options
    primary_button_options: HashMap<String, String>,
    /// The secondary button text
    secondary_button_text: Option<String>,
    /// The secondary button link
    secondary_button_link: Option<String>,
    /// The secondary button options
    secondary_button_options: HashMap<String, String>,
    /// The empty state options
    empty_state_options: HashMap<String, String>,
    /// The description content slot
    description_slot: Option<String>,
}

impl EmptyStateComponent {
    /// Create a new empty state component
    pub fn new(
        compact: bool,
        title: Option<String>,
        svg_path: Option<String>,
        primary_button_text: Option<String>,
        primary_button_link: Option<String>,
        primary_button_options: Option<HashMap<String, String>>,
        secondary_button_text: Option<String>,
        secondary_button_link: Option<String>,
        secondary_button_options: Option<HashMap<String, String>>,
        empty_state_options: Option<HashMap<String, String>>,
    ) -> Self {
        Self {
            compact,
            title,
            svg_path,
            primary_button_text,
            primary_button_link,
            primary_button_options: primary_button_options.unwrap_or_default(),
            secondary_button_text,
            secondary_button_link,
            secondary_button_options: secondary_button_options.unwrap_or_default(),
            empty_state_options: empty_state_options.unwrap_or_default(),
            description_slot: None,
        }
    }

    /// Set the description content slot
    pub fn with_description_slot(mut self, description: String) -> Self {
        self.description_slot = Some(description);
        self
    }

    /// Get the empty state class
    fn empty_state_class(&self) -> String {
        if self.compact {
            "gl-flex-row".to_string()
        } else {
            "gl-text-center gl-flex-col".to_string()
        }
    }

    /// Get the image class
    fn image_class(&self) -> String {
        if self.compact {
            "gl-hidden sm:gl-block gl-px-4".to_string()
        } else {
            "gl-max-w-full".to_string()
        }
    }

    /// Get the content wrapper class
    fn content_wrapper_class(&self) -> String {
        if self.compact {
            "gl-grow gl-basis-0 gl-px-4".to_string()
        } else {
            "gl-m-auto gl-p-5".to_string()
        }
    }

    /// Get the title class
    fn title_class(&self) -> String {
        if self.compact {
            "h5".to_string()
        } else {
            "h4".to_string()
        }
    }

    /// Get the button wrapper class
    fn button_wrapper_class(&self) -> String {
        if self.compact {
            "".to_string()
        } else {
            "gl-justify-center".to_string()
        }
    }

    /// Format the empty state options
    fn formatted_empty_state_options(&self) -> String {
        self.empty_state_options
            .iter()
            .map(|(key, value)| format!(" {}=\"{}\"", key, value))
            .collect::<Vec<String>>()
            .join("")
    }

    /// Render the primary button
    fn render_primary_button(&self) -> Option<String> {
        if let (Some(text), Some(link)) = (&self.primary_button_text, &self.primary_button_link) {
            let mut options = self.primary_button_options.clone();
            options.insert("class".to_string(), "!gl-ml-0".to_string());

            let options_str = options
                .iter()
                .map(|(key, value)| format!(" {}=\"{}\"", key, value))
                .collect::<Vec<String>>()
                .join("");

            Some(format!(
                r#"<a href="{}" class="btn btn-confirm"{}>{}</a>"#,
                link, options_str, text
            ))
        } else {
            None
        }
    }

    /// Render the secondary button
    fn render_secondary_button(&self) -> Option<String> {
        if let (Some(text), Some(link)) = (&self.secondary_button_text, &self.secondary_button_link)
        {
            let mut options = self.secondary_button_options.clone();

            // Add !gl-ml-0 class if primary button is not present
            if self.primary_button_text.is_none() {
                options.insert("class".to_string(), "!gl-ml-0".to_string());
            }

            let options_str = options
                .iter()
                .map(|(key, value)| format!(" {}=\"{}\"", key, value))
                .collect::<Vec<String>>()
                .join("");

            Some(format!(
                r#"<a href="{}" class="btn btn-default"{}>{}</a>"#,
                link, options_str, text
            ))
        } else {
            None
        }
    }

    /// Render the empty state component
    pub fn render(&self) -> String {
        let mut html = String::new();
        let empty_state_class = self.empty_state_class();
        let empty_state_options = self.formatted_empty_state_options();

        // Empty state container
        html.push_str(&format!(
            r#"<section class="gl-flex gl-empty-state {}"{}>"#,
            empty_state_class, empty_state_options
        ));

        // SVG illustration
        if let Some(svg_path) = &self.svg_path {
            let image_class = self.image_class();
            html.push_str(&format!(
                r#"<div class="{}"><img src="{}" alt="" class="gl-dark-invert-keep-hue"></div>"#,
                image_class, svg_path
            ));
        }

        // Content wrapper
        let content_wrapper_class = self.content_wrapper_class();
        html.push_str(&format!(
            r#"<div class="gl-empty-state-content gl-mx-auto gl-my-0 {}">"#,
            content_wrapper_class
        ));

        // Title
        if let Some(title) = &self.title {
            let title_class = self.title_class();
            html.push_str(&format!(
                r#"<h1 class="gl-text-size-h-display gl-leading-36 gl-mt-0 gl-mb-0 {}">{}</h1>"#,
                title_class, title
            ));
        }

        // Description
        if let Some(description) = &self.description_slot {
            html.push_str(&format!(
                r#"<p class="gl-mt-4 gl-mb-0" data-testid="empty-state-description">{}</p>"#,
                description
            ));
        }

        // Buttons
        if self.primary_button_text.is_some() || self.secondary_button_text.is_some() {
            let button_wrapper_class = self.button_wrapper_class();
            html.push_str(&format!(
                r#"<div class="gl-flex gl-flex-wrap gl-mt-5 gl-gap-3 {}">"#,
                button_wrapper_class
            ));

            if let Some(primary_button) = self.render_primary_button() {
                html.push_str(&primary_button);
            }

            if let Some(secondary_button) = self.render_secondary_button() {
                html.push_str(&secondary_button);
            }

            html.push_str("</div>");
        }

        html.push_str("</div></section>");
        html
    }
}
